Natalie Rehfeld
A4 13.4, 13.6

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import math
import statistics
import numpy as np
import scipy.stats
from scipy import stats
from scipy.stats import f_oneway
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm
from statsmodels.formula.api import ols
import statsmodels.stats.multicomp as mc
import matplotlib.pyplot as plt

**13. 4 A process engineer is trying to improve the life of a cutting tool. He has run a 2^3 experiment using a *cutting speed (A)*, *metal hardness (B)*, and *cutting angle (C)* as the factors. The data from the two replicates is in Table 13E.2**

In [3]:
data=pd.read_csv('drive/MyDrive/13.4_Data.csv', sep=',', na_values=".")
data

a  b  c    R
0  -1 -1 -1  221
1   1 -1 -1  325
2  -1  1 -1  354
3   1  1 -1  552
4  -1 -1  1  440
5   1 -1  1  406
6  -1  1  1  605
7   1  1  1  392
8  -1 -1 -1  311
9   1 -1 -1  435
10 -1  1 -1  348
11  1  1 -1  472
12 -1 -1  1  453
13  1 -1  1  377
14 -1  1  1  500
15  1  1  1  419

In [5]:
model = ols('R ~ C(a) + C(b) + C(c) + C(a):C(b) + C(b):C(c) + C(a):C(c) + C(a):C(b):C(c)', data).fit()
aov_table = sm.stats.anova_lm(model, typ=3)
aov_table

sum_sq   df          F    PR(>F)
Intercept       141512.00  1.0  57.466802  0.000064
C(a)             12996.00  1.0   5.277563  0.050684
C(b)              7225.00  1.0   2.934010  0.125086
C(c)             32580.25  1.0  13.230558  0.006614
C(a):C(b)         1104.50  1.0   0.448528  0.521892
C(b):C(c)          220.50  1.0   0.089543  0.772385
C(a):C(c)        14280.50  1.0   5.799188  0.042631
C(a):C(b):C(c)    4830.25  1.0   1.961523  0.198923
Residual         19700.00  8.0        NaN       NaN

**a) Do any of the three factors affect tool life?**

Yes, PR>F for a and c are low numbers compared to all the others. a nd c are cutting angle and speed with are pretty significant

**b) What combination of factor levels produces the longest tool life?**

the best combination for longest tool ife is speed and cutting angle (a and c)

**c) Is there a combination of cutting speed and cutting angle that always gives good results regardless of metal hardness?**

yes, we have one that gives us a value of 0.042631

**13.6 Four factors are thought to possibly influence the taste of a soft-drink beverage: *type of sweetner (A)*, *ratio of syrup to water (B)*, *carbonation level (C)*, and *temperature (D*). Each factor can be run at two levels, producing a 2^4 design. At each run in the design, samples of the beverage are given to a test panel consisting of 20 people. Each tester assigns a point score from 1 to 10 to the beverage. Total score is the response variable, and the objective is to find a formulation that maximizes total score. Two replicates of this design are run, and the results are shown in Table 13E.3. Analyze the data and draw conclusions.**

In [6]:
data1=pd.read_csv('drive/MyDrive/13.6_Data.csv', sep=',', na_values=".")
data1

a  b  c  d    R  Unnamed: 5
0  -1 -1 -1 -1  188         NaN
1   1 -1 -1 -1  172         NaN
2  -1  1 -1 -1  179         NaN
3   1  1 -1 -1  185         NaN
4  -1 -1  1 -1  175         NaN
5   1 -1  1 -1  183         NaN
6  -1  1  1 -1  190         NaN
7   1  1  1 -1  175         NaN
8  -1 -1 -1  1  200         NaN
9   1 -1 -1  1  170         NaN
10 -1  1 -1  1  189         NaN
11  1  1 -1  1  183         NaN
12 -1 -1  1  1  201         NaN
13  1 -1  1  1  181         NaN
14 -1  1  1  1  189         NaN
15  1  1  1  1  178         NaN
16 -1 -1 -1 -1  195         NaN
17  1 -1 -1 -1  180         NaN
18 -1  1 -1 -1  187         NaN
19  1  1 -1 -1  178         NaN
20 -1 -1  1 -1  180         NaN
21  1 -1  1 -1  178         NaN
22 -1  1  1 -1  180         NaN
23  1  1  1 -1  168         NaN
24 -1 -1 -1  1  193         NaN
25  1 -1 -1  1  178         NaN
26 -1  1 -1  1  181         NaN
27  1  1 -1  1  188         NaN
28 -1 -1  1  1  188         NaN
29  1 -1  1  1  173         NaN
30 -1  1  1  1  182         NaN
31  1  1  1  1  182         NaN

In [11]:
df=data1.drop('Unnamed: 5', axis=1)
df

a  b  c  d    R
0  -1 -1 -1 -1  188
1   1 -1 -1 -1  172
2  -1  1 -1 -1  179
3   1  1 -1 -1  185
4  -1 -1  1 -1  175
5   1 -1  1 -1  183
6  -1  1  1 -1  190
7   1  1  1 -1  175
8  -1 -1 -1  1  200
9   1 -1 -1  1  170
10 -1  1 -1  1  189
11  1  1 -1  1  183
12 -1 -1  1  1  201
13  1 -1  1  1  181
14 -1  1  1  1  189
15  1  1  1  1  178
16 -1 -1 -1 -1  195
17  1 -1 -1 -1  180
18 -1  1 -1 -1  187
19  1  1 -1 -1  178
20 -1 -1  1 -1  180
21  1 -1  1 -1  178
22 -1  1  1 -1  180
23  1  1  1 -1  168
24 -1 -1 -1  1  193
25  1 -1 -1  1  178
26 -1  1 -1  1  181
27  1  1 -1  1  188
28 -1 -1  1  1  188
29  1 -1  1  1  173
30 -1  1  1  1  182
31  1  1  1  1  182

In [13]:
df.describe()

a          b          c          d           R
count  32.000000  32.000000  32.000000  32.000000   32.000000
mean    0.000000   0.000000   0.000000   0.000000  182.781250
std     1.016001   1.016001   1.016001   1.016001    7.966601
min    -1.000000  -1.000000  -1.000000  -1.000000  168.000000
25%    -1.000000  -1.000000  -1.000000  -1.000000  178.000000
50%     0.000000   0.000000   0.000000   0.000000  181.500000
75%     1.000000   1.000000   1.000000   1.000000  188.000000
max     1.000000   1.000000   1.000000   1.000000  201.000000

In [15]:
model2 = ols('R ~ C(a) + C(b) + C(c) + C(d) + C(a):C(b) + C(b):C(c) + C(c):C(d) + C(a):C(c) + C(a):C(d) + C(b):C(d) + C(a):C(b):C(c) + C(a):C(b):C(d) + C(a):C(c):C(d) + C(b):C(c):C(d) + C(a):C(b):C(c):C(d)', df).fit()
aov_table2 = sm.stats.anova_lm(model2, typ=3)
aov_table2

sum_sq    df            F        PR(>F)
Intercept            73344.50000   1.0  2537.323243  4.681932e-19
C(a)                   240.25000   1.0     8.311351  1.081710e-02
C(b)                    72.25000   1.0     2.499459  1.334494e-01
C(c)                   196.00000   1.0     6.780541  1.918634e-02
C(d)                    25.00000   1.0     0.864865  3.662006e-01
C(a):C(b)               98.00000   1.0     3.390270  8.419973e-02
C(b):C(c)              128.00000   1.0     4.428108  5.151492e-02
C(c):C(d)               72.00000   1.0     2.490811  1.340769e-01
C(a):C(c)              171.12500   1.0     5.920000  2.707200e-02
C(a):C(d)               24.50000   1.0     0.847568  3.709145e-01
C(b):C(d)                4.50000   1.0     0.155676  6.983790e-01
C(a):C(b):C(c)         232.56250   1.0     8.045405  1.191036e-02
C(a):C(b):C(d)          20.25000   1.0     0.700541  4.149291e-01
C(a):C(c):C(d)          45.56250   1.0     1.576216  2.273330e-01
C(b):C(c):C(d)          45.56250   1.0     1.576216  2.273330e-01
C(a):C(b):C(c):C(d)     47.53125   1.0     1.644324  2.180014e-01
Residual               462.50000  16.0          NaN           NaN

By using a two way factorial anova (with a 95% CL), we can see there is a statistical significance in our data. The first 4 factors show the type of sweetner(.0108) and carbonation level(0.192) make a difference. Looking at different combinations of these factors, we see that sweetner, carbonation level are statistically significant, as well as, ratio of syrup to water and carbonation level. The other factors are not statistically significant and should not be considered.